src/forms/BoundSelectField.test.tsx

```diff
@@ -45,6 +45,23 @@ describe("BoundSelectField", () => {
     );
     expect(r.isAvailable()).toHaveValue("");
   });
+
+  it("has the latest value when onBlur is triggered", async () => {
+    // Given a FormState/ObjectState with an onBlur callback
+    const onBlur = jest.fn();
+    const author: ObjectState<AuthorInput> = createObjectState(
+      formConfig,
+      { favoriteSport: "s:1" },
+      { onBlur: () => onBlur(author.favoriteSport.value) },
+    );
+    const r = await render(<BoundSelectField field={author.favoriteSport} options={sports} />);
+    // When changing the value
+    r.favoriteSport().focus();
+    click(r.getByRole("option", { name: "Soccer" }));
+
+    // Then formState has the latest value when onBlur is called
+    expect(onBlur).toBeCalledWith("s:2");
```

Comment:
Ah nice! Thanks!

Really nice how you were able to black-box test this, i.e. we don't need to worry too much about poking around internal implementation details/order of events/etc., and instead just observe "yes the `.value` was correct at time of blur".